language/tools/move-cli/src/package/cli.rs

```diff
@@ -360,115 +360,131 @@ pub fn handle_package_commands(
-                                .insert(module.module.self_id(), module.clone());
-                        }
-                    }
+    let (test_plan, mut files, units) = test_plan.unwrap();
+    files.extend(dep_file_map);
+    let test_plan = test_plan.unwrap();
+    let no_tests = test_plan.is_empty();
+    let mut test_plan = TestPlan::new(test_plan, files, units);
+    for pkg in pkg.0.transitive_dependencies() {
+        for unit in &pkg.compiled_units {
+            match &unit.unit {
+                CompiledUnit::Script(_) => (),
+                CompiledUnit::Module(module) => {
+                    test_plan
+                        .module_info
+                        .insert(module.module.self_id(), module.clone());
                 }
             }
+        }
+    }
 
-            let trace_path = rooted_path.join(".trace");
-            let coverage_map_path = rooted_path
-                .join(".coverage_map")
-                .with_extension(MOVE_COVERAGE_MAP_EXTENSION);
-            let cleanup_trace = || {
-                if *compute_coverage && trace_path.exists() {
-                    std::fs::remove_file(&trace_path).unwrap();
-                }
-            };
+    let trace_path = pkg_path.join(".trace");
+    let coverage_map_path = pkg_path
+        .join(".coverage_map")
+        .with_extension(MOVE_COVERAGE_MAP_EXTENSION);
+    let cleanup_trace = || {
+        if compute_coverage && trace_path.exists() {
+            std::fs::remove_file(&trace_path).unwrap();
+        }
+    };
 
-            cleanup_trace();
+    cleanup_trace();
 
-            if *compute_coverage {
-                std::env::set_var("MOVE_VM_TRACE", &trace_path);
-            }
+    if compute_coverage {
+        std::env::set_var("MOVE_VM_TRACE", &trace_path);
+    }
 
-            if !unit_test_config
-                .run_and_report_unit_tests(test_plan, Some(natives), std::io::stdout())
-                .unwrap()
-                .1
-            {
-                cleanup_trace();
-                std::process::exit(1)
-            }
+    if !unit_test_config
+        .run_and_report_unit_tests(test_plan, Some(natives), std::io::stdout())
+        .unwrap()
+        .1
+    {
+        cleanup_trace();
+        return Ok(UnitTestResult::Failure);
+    }
 
-            if *compute_coverage && !no_tests {
-                let coverage_map = CoverageMap::from_trace_file(trace_path);
-                output_map_to_file(&coverage_map_path, &coverage_map).unwrap();
-            }
+    if compute_coverage && !no_tests {
+        let coverage_map = CoverageMap::from_trace_file(trace_path);
+        output_map_to_file(&coverage_map_path, &coverage_map).unwrap();
+    }
+    Ok(UnitTestResult::Success)
+}
 
-            std::process::exit(0)
-        }
-        PackageCommand::CoverageReport { options } => {
-            options.handle_command(config, &rooted_path?)?;
-        }
-    };
-    Ok(())
+pub enum UnitTestResult {
```

Comment:
Nit: I think we generally try to keep enums/structs near/at the top of the file. Could you possibly move this to the top of this file?